src/plugins/walletConnectToDapps/components/DappRegistryGrid.tsx

```diff
@@ -0,0 +1,96 @@
+import { SearchIcon } from '@chakra-ui/icons'
+import {
+  Box,
+  Heading,
+  Image,
+  Input,
+  InputGroup,
+  InputLeftElement,
+  Link,
+  SimpleGrid,
+  Stack,
+  Text as PlainText
+} from '@chakra-ui/react'
+import { Text } from 'components/Text'
+import { FC, useEffect, useMemo } from 'react'
+import { useForm } from 'react-hook-form'
+
+interface RegistryItem {
+  category: string
+  id: string
+  homepage: string
+  name: string
+  image: string
+}
+
+const registry: RegistryItem[] = require('../registry.json')
+
+const pageSize = 20
+
+export const DappRegistryGrid: FC = () => {
+  const { register, watch, setValue } = useForm<{ search: string; page: number }>({
+    mode: 'onChange',
+    defaultValues: {
+      search: '',
+      page: 0,
+    },
+  })
+
+  const search = watch('search')
+  const page = watch('page')
+  useEffect(() => setValue('page', 0), [search, setValue])
+
+  const filteredListings = useMemo(
+    () => registry.filter(l => !search || l.name.toLowerCase().includes(search.toLowerCase())),
+    [search],
+  )
+
+  return (
+    <Box>
+      <Stack direction='row' alignItems='center' mb={4}>
+        <Heading flex={1} fontSize='2xl'>
+          <Text translation='plugins.walletConnectToDapps.availableDapps' />
+        </Heading>
+        <Box>
+          <InputGroup>
+            <InputLeftElement pointerEvents='none'>
+              <SearchIcon color='gray.700' />
+            </InputLeftElement>
+            <Input
+              {...register('search')}
+              type='text'
+              placeholder='Search'
+              pl={10}
+              variant='filled'
+            />
+          </InputGroup>
+        </Box>
+      </Stack>
+      <SimpleGrid columns={{ lg: 4, sm: 2, base: 1 }} spacing={4}>
+        {filteredListings.slice(page * pageSize, pageSize).map(listing => (
+          <Link key={listing.id} href={listing.homepage} isExternal>
```

Comment:
@shapeshift/product are we happy with this hover style?

<img width="255" alt="image" src="https://user-images.githubusercontent.com/17035424/189231977-0acbb3d2-94c9-45a9-bbab-22a127a24744.png">

Reply:
If possible it would be good to increase the opacity of the blurred background image on hover from 0.3 to 0.5. I'm not against the underline for now.

Reply:
Done